kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,911 @@
+        // Set flag
+        this->Set(NODES_CONTAINER_INITIALIZED, true);
+    }
+
+    return mNodesArray;
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+FromJSONCheckResultProcess::ElementsArrayType& FromJSONCheckResultProcess::GetElements(const Flags* pFlag)
+{
+    KRATOS_TRY;
+
+    if (this->IsNot(ELEMENTS_CONTAINER_INITIALIZED)) {
+        auto& r_elements_array = this->Is(CHECK_ONLY_LOCAL_ENTITIES) ? mrModelPart.GetCommunicator().LocalMesh().Elements() : mrModelPart.Elements();
+        const auto it_elem_begin = r_elements_array.begin();
+        const int number_of_elements = static_cast<int>(r_elements_array.size());
+
+        #pragma omp parallel firstprivate(number_of_elements)
+        {
+            ElementsArrayType elements_buffer;
+
+            # pragma omp for schedule(guided, 512) nowait
+            for (int i = 0; i < number_of_elements; ++i) {
+                auto it_elem = it_elem_begin + i;
+                if (CheckFlag(*it_elem, pFlag)) elements_buffer.push_back(*(it_elem.base()));
+
+            }
+
+            // We merge all the sets in one thread
+            #pragma omp critical
+            {
+                for (auto it_elem = elements_buffer.begin(); it_elem < elements_buffer.end(); ++it_elem)
+                    mElementsArray.push_back(*(it_elem.base()));
+            }
+        }
+        // Set flag
+        this->Set(ELEMENTS_CONTAINER_INITIALIZED, true);
+    }
+
+    return mElementsArray;
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+std::size_t FromJSONCheckResultProcess::ComputeRelevantDigits(const double Value)
+{
+    return static_cast<std::size_t>(std::ceil(std::abs(std::log10(Value))));
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+const Parameters FromJSONCheckResultProcess::GetDefaultParameters() const
+{
+    const Parameters default_parameters = Parameters(R"(
+    {
+        "help"                         : "This process checks the solution obtained from a given json file. It can be used for generating tests for a problem",
+        "check_variables"              : [],
+        "gauss_points_check_variables" : [],
+        "input_file_name"              : "",
+        "model_part_name"              : "",
+        "sub_model_part_name"          : "",
+        "check_for_flag"               : "",
+        "historical_value"             : true,
+        "tolerance"                    : 1e-3,
+        "relative_tolerance"           : 1e-6,
+        "time_frequency"               : 1.00,
+        "use_node_coordinates"         : false,
```

Comment:
this is no longer needed
I forgot to deprecate this

Was only needed in MPI because the IDs were not consistent with the serial case, but this was solved

Reply:
I will keep it in this PR, otherwise this PR would include test modifications, and I want to keep it as a direct replace

Reply:
can you please add a deprecation warning then?

I am pretty sure I am the only one using this
Once I updated my tests I will then remove it